posthog/api/survey.py

```diff
@@ -379,7 +379,11 @@ def update(self, instance: Survey, validated_data):
             instance.targeting_flag.save()
 
         iteration_count = validated_data.get("iteration_count")
-        if instance.current_iteration is not None and instance.current_iteration > iteration_count > 0:
+        if (
+            instance.current_iteration is not None
+            and iteration_count is not None
+            and instance.current_iteration > iteration_count > 0
```

Comment:
Okay, so this was the fix I was going to go with, but the reason I didn't just fix this myself is because this doesn't tell me _why_ these illegal states are possible?  Like, if we have a situation where we have a `current_iteration` but no `iteration_count`, isn't that something we should make impossible or force-correct?  can you tell me more about how this state could be reached?

Reply:
This is an invalid state, where the survey previously had recurrence set on it and went to current_iteration of 2.
Then recurrence was disabled and customer tried to set it to be recurring again. Let me write a test for the full scenario.

Reply:
Fixed in https://github.com/PostHog/posthog/pull/23917/commits/14d2f5a636e32d0c5e70af5a2aa1e1f836c85a13#diff-c2c811931c76d29d59ee8c932d3ef20d484ecffe1fa1ca049f6abae7edf437a2R179-R184